File: Algorithms/633. Sum of Square Numbers.rs
impl Solution {
    pub fn judge_square_sum(c: i32) -> bool {
        if c == 0 {
            return true;
        }
        let x = (c as f64).sqrt() as i32;
        for i in 0..=x {
            let temp = c - i * i;
            let test = (temp as f64).sqrt() as i32;
            if temp == test * test {
                return true;
            }
        }
        false
    }
}

File: Algorithms/633. Sum of Square Numbers.ts
function judgeSquareSum(c: number): boolean {
  let [a, b] = [0, Math.floor(Math.sqrt(c))];
  while (a <= b) {
    const s = a * a + b * b;
    if (s === c) {
      return true;
    }
    if (s < c) {
      a += 1;
    } else {
      b -= 1;
    }
  }
  return false;
}
